packages/s/stduuid/xmake.lua

```diff
@@ -26,13 +29,21 @@ package("stduuid")
         if package:config("time") then
             package:add("defines", "UUID_TIME_GENERATOR")
         end
+        if not package:config("span") then
+            package:add("deps", "microsoft-gsl")
+        end
     end)
 
     on_install(function (package)
+        io.replace("CMakeLists.txt", "install(FILES include/uuid.h DESTINATION include)", "install(FILES include/uuid.h DESTINATION include/stduuid)", {plain = true})
+
         local configs = {"-DUUID_BUILD_TESTS=OFF", "-DUUID_ENABLE_INSTALL=ON"}
         table.insert(configs, "-DCMAKE_BUILD_TYPE=" .. (package:is_debug() and "Debug" or "Release"))
-        table.insert(configs, "-DBUILD_SHARED_LIBS=" .. (package:config("shared") and "ON" or "OFF"))
+        table.insert(configs, "-DUUID_USING_CXX20_SPAN=" .. (package:config("span") and "ON" or "OFF"))
```

Comment:
when enabling this, I guess C++20 is required, but the on_test part only enables C++17, I guess we should switch to C++20 if span is enabled